### ISSUE TITLE
Overflow Exception In Trap Handling

### ISSUE THREAD

**Issue by martindevans** [bytecodealliance/wasmtime-dotnet#150]
I'm experimenting with a better way to handle traps and I've come across what looks like a bug in the current implementation.

I modified `Trap.wat` to cause a stackoverflow instead of using `unreachable`. I expected that to cause a `TrapException` of type `StackOverflow`, instead I'm seeing an `OverflowException`:

```csharp
Xunit.Sdk.XunitException
Expected a <Wasmtime.TrapException> to be thrown, but found <System.OverflowException>: "
"System.OverflowException with message "Arithmetic operation resulted in an overflow."
     at System.UIntPtr.op_Explicit(UIntPtr value)
     at Wasmtime.TrapFrame..ctor(IntPtr frame) in C:\Users\Martin\Documents\dotnet\wasmtime-dotnet\src\TrapException.cs:line 50
     at Wasmtime.TrapException.FromOwnedTrap(IntPtr trap) in C:\Users\Martin\Documents\dotnet\wasmtime-dotnet\src\TrapException.cs:line 179
     at Wasmtime.Function.Invoke(ReadOnlySpan`1 arguments, Span`1 resultsOut) in C:\Users\Martin\Documents\dotnet\wasmtime-dotnet\src\Function.cs:line 2218
     at Wasmtime.Function.InvokeWithoutReturn(ReadOnlySpan`1 arguments) in C:\Users\Martin\Documents\dotnet\wasmtime-dotnet\src\Function.cs:line 2063
```

This is coming from this code in the `TrapFrame` constructor:

```csharp
FunctionOffset = (int)Native.wasm_frame_func_offset(frame); // <-- Exception Here!
FunctionName = null;
ModuleOffset = (int)Native.wasm_frame_module_offset(frame);
ModuleName = null;
```

`Native.wasm_frame_func_offset` returns a `UIntPtr` (platform dependent, 64 bits on my PC) but it's being cast into an `int` (always 32 bits).

Should `FunctionOffset` be converted into an `nuint` instead? If so should the same change be made to `ModuleName`?

**Comment by martindevans** [comment]
The overflow exception was fixed in #153, which included the change to use `nuint` instead (see [here](https://github.com/bytecodealliance/wasmtime-dotnet/blob/main/src/TrapException.cs#L71)).

**Comment by peterhuene** [comment]
Thanks for verifying this!
